### PR TITLE
Update stack versions

### PR DESCRIPTION
**Pull request by barkbay**
This PR updates the stack versions in our recipes, samples and E2E tests/pipelines.

### CODE REVIEW

File: .ci/pipelines/e2e-tests-stack-versions-gke.Jenkinsfile [hunk 1]
```diff
@@ -141,18 +141,29 @@ pipeline {
...
                     }
                     steps {
                         unstash "source"
                         script {
-                            runWith(lib, failedTests, "eck-820-${BUILD_NUMBER}-e2e", "8.2.2")
+                            runWith(lib, failedTests, "eck-82-${BUILD_NUMBER}-e2e", "8.2.3")
+                        }
+                    }
+               }
+               stage("8.3.0") {
```
Comment (pebrc) [inline .ci/pipelines/e2e-tests-stack-versions-gke.Jenkinsfile:159]: Should we remove one of the older 7.x versions from testing to get the overall number of tests constant?

**Review by pebrc — APPROVED** [review]

File: test/e2e/test/version.go [hunk 2]
```diff
@@ -16,9 +16,9 @@ const (
 	// LatestReleasedVersion6x is the latest released version for 6.x
 	LatestReleasedVersion6x = "6.8.23"
 	// LatestReleasedVersion7x is the latest released version for 7.x
-	LatestReleasedVersion7x = "7.17.4"
+	LatestReleasedVersion7x = "7.17.5"
```
Comment (thbkrkr) [inline test/e2e/test/version.go:19]: There is also a `7.17.x` to update here: https://github.com/elastic/cloud-on-k8s/blob/d23142646ff9e7af1bcb4d6a1a7150ee245ae398/.ci/pipelines/e2e-tests-stack-versions-gke.Jenkinsfile#L104

**Review by thbkrkr — APPROVED** [review]

**Comment by barkbay** [comment]
run/e2e-tests

**Comment by barkbay** [comment]
run/e2e-tests

**Comment by barkbay** [comment]
run/e2e-tests

**Comment by barkbay** [comment]
run/e2e-tests

**Comment by barkbay** [comment]
run/e2e-tests

**Comment by barkbay** [comment]
run/e2e-tests

**Comment by barkbay** [comment]
image
🎉

**Pull request merged**